modules/storage-aws-s3/s3/s3.go

```diff
@@ -102,8 +111,56 @@ func (s *s3) StoreSnapshot(ctx context.Context, snapshot snapshots.Snapshot) err
 	return nil
 }
 
+func (s *s3) makeFilePath(dataPath, relPath string) string {
+	return fmt.Sprintf("%s/%s", dataPath, relPath)
+}
+
+func (s *s3) makeObjectName(snapshotId, srcRelPath string) string {
+	return fmt.Sprintf("%s/%s", snapshotId, srcRelPath)
+}
+
 func (s *s3) RestoreSnapshot(ctx context.Context, snapshotId string) error {
-	// TODO implement
-	s.logger.Errorf("RestoreSnapshot of StorageAWSS3Module not yet implemented")
+	bucketName := s.config.BucketName()
+	bucketExists, err := s.client.BucketExists(ctx, bucketName)
+	if !bucketExists {
+		return errors.Wrap(err, "backup bucket does not exist")
+	}
+	if err != nil {
+		return errors.Wrap(err, "can't connect to bucket")
+	}
+
+	objectName := fmt.Sprintf("%s/%s", snapshotId, "snapshot.json")
```

Comment:
it would seem like a call to `func (s *s3) makeObjectName` would be warranted here, but as mentioned above, `path.Join` will do this too.